app/models/beneficiaries/IndividualBeneficiary.scala

```diff
@@ -40,32 +43,42 @@ object IndividualBeneficiary extends BeneficiaryReads {
     ).tupled.map{
-    case (name, dob, nino, identification, vulnerable, employment, None, _, entityStart, provisional) =>
-      IndividualBeneficiary(name, dob, nino, identification, vulnerable, employment, None, incomeDiscretionYesNo = true, entityStart, provisional)
-    case (name, dob, nino, identification, vulnerable, employment, _, Some(true), entityStart, provisional) =>
-      IndividualBeneficiary(name, dob, nino, identification, vulnerable, employment, None, incomeDiscretionYesNo = true, entityStart, provisional)
-    case (name, dob, nino, identification, vulnerable,  employment, income, _, entityStart, provisional) =>
-      IndividualBeneficiary(name, dob, nino, identification, vulnerable, employment, income, incomeDiscretionYesNo = false, entityStart, provisional)
+    case (name, dob, None, None, None, None, None, None, country, nationality, legallyIncapable, entityStart, provisional) =>
+      IndividualBeneficiary(name, dob, None, None, None, None, None, None, country, nationality, legallyIncapable.map(!_), entityStart, provisional)
+    case (name, dob, nino, identification, vulnerable, employment, None, _, country, nationality, legallyIncapable, entityStart, provisional) =>
+      IndividualBeneficiary(name, dob, nino, identification, vulnerable, employment, None, incomeDiscretionYesNo = Some(true), country, nationality, legallyIncapable.map(!_), entityStart, provisional)
+    case (name, dob, nino, identification, vulnerable, employment, _, Some(true), country, nationality, legallyIncapable, entityStart, provisional) =>
+      IndividualBeneficiary(name, dob, nino, identification, vulnerable, employment, None, incomeDiscretionYesNo = Some(true), country, nationality, legallyIncapable.map(!_), entityStart, provisional)
+    case (name, dob, nino, identification, vulnerable,  employment, income, _, country, nationality, legallyIncapable, entityStart, provisional) =>
+      IndividualBeneficiary(name, dob, nino, identification, vulnerable, employment, income, incomeDiscretionYesNo = Some(false), country, nationality, legallyIncapable.map(!_), entityStart, provisional)
   }
 
   implicit val writes: Writes[IndividualBeneficiary] = (
     (__ \ 'name).write[Name] and
       (__ \ 'dateOfBirth).writeNullable[LocalDate] and
       (__ \ 'identification).writeNullable[IndividualIdentification] and
       (__ \ 'identification \ 'address).writeNullable[Address] and
-      (__ \ 'vulnerableBeneficiary).write[Boolean] and
+      (__ \ 'vulnerableBeneficiary).writeNullable[Boolean] and
       (__ \ 'beneficiaryType).writeNullable[RoleInCompany] and
       (__ \ 'beneficiaryShareOfIncome).writeNullable[String] and
-      (__ \ 'beneficiaryDiscretion).write[Boolean] and
+      (__ \ 'beneficiaryDiscretion).writeNullable[Boolean] and
+      (__ \ 'countryOfResidence).writeNullable[String] and
+      (__ \ 'nationality).writeNullable[String] and
+      (__ \ 'legallyIncapable).writeNullable[Boolean] and
```

Comment:
So I _think_ we can make this `(__ \ 'mentalCapacity).writeNullable[Boolean](Writes[Boolean](x => JsBoolean(!x)))` and then revert the changes below

Reply:
Updated